### DISCUSSION TITLE
Command Timeouts

### DISCUSSION THREAD

**Discussion by jbowser-bt** [Azure/data-api-builder#2275] [General]
Has there been internal discussion around exposing timeouts for querying data? It seems as though the default SqlCommand timeout of 30 seconds is in play. Has there been discussion about exposing this and making it configurable?

**Comment by sezal98** [comment]
Hi @jbowser-bt , Can you elaborate a little on your use case

**Reply by jbowser-bt** [reply]
Not sure how I missed this. I'm using DAB as a no-code solution to expose an API for an Azure SQL Database. 

I'm hitting issues where there are certain queries that take between 27-33 seconds to run. If they hit 30 seconds, DAB returns a timeout error. It happens that 30 seconds is the default timeout for SqlCommand. I'm hoping to extend the timeout for these queries against my Azure SQL Database for a more reliable user experience. Unfortunately, the Command Timeout parameter for a connection string does not apply to an Azure SQL Database connection.

Due to the complexity of the data store, and it having not been designed for data reporting/querying, I'm running out of levers I can pull at the database level to improve performance of the queries themselves to keep it under 30 seconds reliably.

**Comment by JerryNixon** [comment]
Added to the docs.
